core/src/main/scalajvm/sttp/tapir/static/Files.scala

```diff
@@ -32,37 +34,52 @@ object Files {
 
   private def files[F[_]](realSystemPath: Path, calculateETag: File => F[Option[ETag]])(filesInput: StaticInput)(implicit
       m: MonadError[F]
-  ): F[Either[StaticErrorOutput, StaticOutput[File]]] = {
+  ): F[Either[StaticErrorOutput, StaticOutput[TapirFile]]] = {
     val resolved = filesInput.path.foldLeft(realSystemPath)(_.resolve(_))
     m.flatten(m.blocking {
       if (!java.nio.file.Files.exists(resolved, LinkOption.NOFOLLOW_LINKS))
-        (Left(StaticErrorOutput.NotFound): Either[StaticErrorOutput, StaticOutput[File]]).unit
+        (Left(StaticErrorOutput.NotFound): Either[StaticErrorOutput, StaticOutput[TapirFile]]).unit
       else {
         val realRequestedPath = resolved.toRealPath(LinkOption.NOFOLLOW_LINKS)
         if (!realRequestedPath.startsWith(realSystemPath))
-          (Left(StaticErrorOutput.NotFound): Either[StaticErrorOutput, StaticOutput[File]]).unit
+          (Left(StaticErrorOutput.NotFound): Either[StaticErrorOutput, StaticOutput[TapirFile]]).unit
         else if (realRequestedPath.toFile.isDirectory) {
           files(realSystemPath, calculateETag)(filesInput.copy(path = filesInput.path :+ "index.html"))
-        } else fileOutput(filesInput, realRequestedPath, calculateETag).map(Right(_))
+        } else {
+          filesInput.range match {
+            case Some(range) =>
+              val file = realRequestedPath.toFile
+              if (range.validate(file.length())) fileOutput(filesInput, file, calculateETag, range).map(Right(_))
```

Comment:
shouldn't we respond with `206 Partial Content` in case of a range request? https://developer.mozilla.org/en-US/docs/Web/HTTP/Range_requests